tests are failing, the issue is related to the directive interface 
[build error](https://github.com/persian-tools/vue-persian-tools/pull/1/checks?check_run_id=2426024897)
```
 TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/directives/halfSpace/__tests__/halfSpaceDirective.spec.ts:15:31 - error TS2769: No overload matches this call.
      The last overload gave the following error.
        Argument of type '{ props: string[]; directives: { halfSpace: Directive; }; template: string; }' is not assignable to parameter of type 'ComponentOptionsWithObjectProps<string[], unknown, unknown, {}, {}, Record<string, any>, ComponentOptionsMixin, ComponentOptionsMixin, string, Readonly<{ [x: number]: string; } & { ...; }>, { ...; }>'.
          Type '{ props: string[]; directives: { halfSpace: Directive; }; template: string; }' is not assignable to type 'ComponentOptionsBase<Readonly<{ [x: number]: string; } & { length?: number | undefined; toString?: string | undefined; toLocaleString?: string | undefined; pop?: string | undefined; push?: number | undefined; concat?: string[] | undefined; ... 25 more ...; flat?: unknown[] | undefined; }>, ... 8 more ..., { ...; }>'.
            Types of property 'directives' are incompatible.
              Type '{ halfSpace: Directive; }' is not assignable to type 'Record<string, Directive<any, any>>'.
                Property 'halfSpace' is incompatible with index signature.
                  Type 'Directive' is not assignable to type 'Directive<any, any>'.
                    Type 'Vue2Directive<any>' is not assignable to type 'Directive<any, any>'.
                      Type 'Vue2Directive<any>' is not assignable to type 'DirectiveHook<any, any, any>'.
                        Type 'Vue2Directive<any>' provides no match for the signature '(el: any, binding: DirectiveBinding<any>, vnode: VNode<any, any, { [key: string]: any; }>, prevVNode: any): void'.

    15         const wrapper = mount(cmp, {
                                     ~~~

      node_modules/@vue/test-utils/dist/mount.d.ts:25:25
        25 export declare function mount<PropsOptions extends Readonly<ComponentPropsOptions>, RawBindings, D, C extends ComputedOptions = {}, M extends Record<string, Function> = {}, E extends EmitsOptions = Record<string, any>, Mixin extends ComponentOptionsMixin = ComponentOptionsMixin, Extends extends ComponentOptionsMixin = ComponentOptionsMixin, EE extends string = string>(componentOptions: ComponentOptionsWithObjectProps<PropsOptions, RawBindings, D, C, M, E, Mixin, Extends, EE>, options?: MountingOptions<ExtractPropTypes<PropsOptions> & PublicProps, D>): VueWrapper<ComponentPublicInstance<ExtractPropTypes<PropsOptions>, RawBindings, D, C, M, E, VNodeProps & ExtractPropTypes<PropsOptions>>>;
                                   ~~~~~
        The last overload is declared here.
    src/directives/halfSpace/__tests__/halfSpaceDirective.spec.ts:24:31 - error TS2769: No overload matches this call.
      The last overload gave the following error.
        Argument of type '{ props: string[]; directives: { halfSpace: Directive; }; template: string; }' is not assignable to parameter of type 'ComponentOptionsWithObjectProps<string[], unknown, unknown, {}, {}, Record<string, any>, ComponentOptionsMixin, ComponentOptionsMixin, string, Readonly<{ [x: number]: string; } & { ...; }>, { ...; }>'.
          Type '{ props: string[]; directives: { halfSpace: Directive; }; template: string; }' is not assignable to type 'ComponentOptionsBase<Readonly<{ [x: number]: string; } & { length?: number | undefined; toString?: string | undefined; toLocaleString?: string | undefined; pop?: string | undefined; push?: number | undefined; concat?: string[] | undefined; ... 25 more ...; flat?: unknown[] | undefined; }>, ... 8 more ..., { ...; }>'.

    24         const wrapper = mount(cmp);
                                     ~~~

      node_modules/@vue/test-utils/dist/mount.d.ts:25:25
        25 export declare function mount<PropsOptions extends Readonly<ComponentPropsOptions>, RawBindings, D, C extends ComputedOptions = {}, M extends Record<string, Function> = {}, E extends EmitsOptions = Record<string, any>, Mixin extends ComponentOptionsMixin = ComponentOptionsMixin, Extends extends ComponentOptionsMixin = ComponentOptionsMixin, EE extends string = string>(componentOptions: ComponentOptionsWithObjectProps<PropsOptions, RawBindings, D, C, M, E, Mixin, Extends, EE>, options?: MountingOptions<ExtractPropTypes<PropsOptions> & PublicProps, D>): VueWrapper<ComponentPublicInstance<ExtractPropTypes<PropsOptions>, RawBindings, D, C, M, E, VNodeProps & ExtractPropTypes<PropsOptions>>>;
                                   ~~~~~
        The last overload is declared here.
    src/directives/halfSpace/__tests__/halfSpaceDirective.spec.ts:38:31 - error TS2769: No overload matches this call.
      The last overload gave the following error.
        Argument of type '{ directives: { halfSpace: Directive; }; template: string; }' is not assignable to parameter of type 'ComponentOptionsWithObjectProps<readonly string[] | Readonly<ComponentObjectPropsOptions<Record<string, unknown>>>, unknown, unknown, {}, {}, Record<string, any>, ComponentOptionsMixin, ComponentOptionsMixin, string, Readonly<...> | Readonly<...>, { ...; } | {}>'.
          Type '{ directives: { halfSpace: Directive; }; template: string; }' is not assignable to type 'ComponentOptionsBase<Readonly<{ [x: number]: string; } & { length?: number | undefined; toString?: string | undefined; toLocaleString?: string | undefined; concat?: string[] | undefined; join?: string | undefined; ... 17 more ...; flat?: unknown[] | undefined; }> | Readonly<...>, ... 8 more ..., { ...; } | {}>'.
            Types of property 'directives' are incompatible.
              Type '{ halfSpace: Directive; }' is not assignable to type 'Record<string, Directive<any, any>>'.
                Property 'halfSpace' is incompatible with index signature.
                  Type 'Directive' is not assignable to type 'Directive<any, any>'.
                    Type 'Vue2Directive<any>' is not assignable to type 'Directive<any, any>'.
                      Type 'Vue2Directive<any>' is not assignable to type 'DirectiveHook<any, any, any>'.

    38         const wrapper = mount(cmpSync);
                                     ~~~~~~~

      node_modules/@vue/test-utils/dist/mount.d.ts:25:25
        25 export declare function mount<PropsOptions extends Readonly<ComponentPropsOptions>, RawBindings, D, C extends ComputedOptions = {}, M extends Record<string, Function> = {}, E extends EmitsOptions = Record<string, any>, Mixin extends ComponentOptionsMixin = ComponentOptionsMixin, Extends extends ComponentOptionsMixin = ComponentOptionsMixin, EE extends string = string>(componentOptions: ComponentOptionsWithObjectProps<PropsOptions, RawBindings, D, C, M, E, Mixin, Extends, EE>, options?: MountingOptions<ExtractPropTypes<PropsOptions> & PublicProps, D>): VueWrapper<ComponentPublicInstance<ExtractPropTypes<PropsOptions>, RawBindings, D, C, M, E, VNodeProps & ExtractPropTypes<PropsOptions>>>;
                                   ~~~~~
        The last overload is declared here.
```

--- a/src/directives/helpers/utils.ts
+++ b/src/directives/helpers/utils.ts
@@ -3,12 +3,12 @@ import { ObjectDirective as Vue3Directive, DirectiveBinding as Vue3DirectiveBind
 interface Vue2DirectiveBinding {
     modifiers: Record<string, boolean>;
 }
-interface Vue2Directive<T = any> {
+interface Vue2Directive<T = HTMLInputElement> {
     bind?: (el: T, binding: Vue2DirectiveBinding) => void;
     unbind?: (el: T, binding: Vue2DirectiveBinding) => void;
 }
 
-type Directive = Vue2Directive | Vue3Directive;
+interface Directive<T = HTMLInputElement> extends Vue2Directive<T>, Vue3Directive {}
 type DirectiveBinding = Vue2DirectiveBinding | Vue3DirectiveBinding;
 
-export { Vue2DirectiveBinding, Vue2Directive, Vue3Directive, Directive, DirectiveBinding };
+export { Vue2DirectiveBinding, Vue2Directive, Vue3DirectiveBinding, Vue3Directive, Directive, DirectiveBinding };

